Sources/FoundationEssentials/ProcessInfo/ProcessInfo.swift

```diff
@@ -203,10 +202,12 @@ final class _ProcessInfo: Sendable {
         return ""
 #elseif os(Windows)
         var ulLength: ULONG = 0
-        GetUserNameExW(NameDisplay, NULL, &ulLength)
+        _ = GetUserNameExW(NameDisplay, nil, &ulLength)
 
-        return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: ulLength + 1) { wszBuffer in
-            GetUserNameExW(NameDisplay, wszBuffer.baseAddress!, &ulLength)
+        return withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(ulLength + 1)) { wszBuffer in
```

Comment:
From the [docs](https://learn.microsoft.com/en-us/windows/win32/api/secext/nf-secext-getusernameexw) I see:

> If lpNameBuffer is too small, the function fails and [GetLastError](https://learn.microsoft.com/en-us/windows/desktop/api/errhandlingapi/nf-errhandlingapi-getlasterror) returns ERROR_MORE_DATA. This parameter receives the required buffer size, in Unicode characters (whether or not Unicode is being used), including the terminating null character.

I think that means we can allocate `length` instead of `ulLength + 1` right? (and we probably also need to use `ulLength - 1` below)

Reply:
Ah, right! I didn't cross-reference that.